src/interpreter/ByteCodeGenerator.cpp

```diff
@@ -88,14 +88,34 @@ void ByteCodeGenerateContext::morphJumpPositionIntoComplexCase(ByteCodeBlock* cb
 }
 
 #ifdef ESCARGOT_DEBUGGER
-void ByteCodeGenerateContext::insertBreakpoint(size_t line, Node* node)
+void ByteCodeGenerateContext::insertBreakpoint(size_t index, Node* node)
 {
     ASSERT(m_breakpointContext != nullptr);
+    ASSERT (index != SIZE_MAX);
+
+    StringView src = m_codeBlock->asInterpretedCodeBlock()->src();
+    ExtendedNodeLOC sourceElementStart = m_codeBlock->asInterpretedCodeBlock()->functionStart();
+    size_t lastLine = m_breakpointContext->m_lastBreakpointLine;
+    index -= sourceElementStart.index;
+
+    if (index > m_breakpointContext->m_lastBreakpointIndex) {
+        for (size_t i = m_breakpointContext->m_lastBreakpointIndex; i < index; i++) {
+            char16_t c = src.charAt(i);
+            if (EscargotLexer::isLineTerminator(c)) {
+                // skip \r\n
+                if (UNLIKELY (c == 13 && (i + 1 < index) && src.charAt(i + 1) == 10)) {
```

Comment:
Don't we have any constants for newline?

Reply:
No we don't.